rust-src/concordium_base/src/id/sigma_protocols/aggregate_dlog.rs

```diff
@@ -60,37 +57,41 @@ impl<C: Curve> SigmaProtocol for AggregateDlog<C> {
         Some((multiexp(&self.coeff, &rands), rands))
     }
 
-    fn generate_witness(
+    fn get_challenge(&self, challenge: &Challenge) -> Self::ProtocolChallenge {
+        C::scalar_from_bytes(challenge)
+    }
+
+    fn compute_response(
         &self,
         secret: Self::SecretData,
         state: Self::ProverState,
         challenge: &Self::ProtocolChallenge,
-    ) -> Option<Self::ProverWitness> {
+    ) -> Option<Self::Response> {
         let n = secret.len();
         if state.len() != n {
             return None;
         }
-        let mut witness = Vec::with_capacity(n);
+        let mut response = Vec::with_capacity(n);
         for (ref s, ref r) in izip!(secret, state) {
-            let mut wit = *challenge;
-            wit.mul_assign(s);
-            wit.negate();
-            wit.add_assign(r);
-            witness.push(wit);
+            let mut res = *challenge;
+            res.mul_assign(s);
+            res.negate();
+            res.add_assign(r);
+            response.push(res);
         }
-        Some(Witness { witness })
+        Some(Response { response })
     }
 
-    fn extract_point(
+    fn extract_commit_message(
         &self,
         challenge: &Self::ProtocolChallenge,
-        witness: &Self::ProverWitness,
+        response: &Self::Response,
     ) -> Option<Self::CommitMessage> {
-        if witness.witness.len() != self.coeff.len() {
+        if response.response.len() != self.coeff.len() {
             return None;
         }
         let mut point = self.public.mul_by_scalar(challenge);
```

Comment:
Maybe we can make the code clearer by renaming `point` to `commit_message` (although `point` is also fine)

Reply:
Make sense.